Ternary Search Tree/Tests/TernarySearchTreeTests.swift

```diff
@@ -10,14 +10,23 @@ import XCTest
 
 class TernarySearchTreeTests: XCTestCase {
     
-    let treeOfStrings = TernarySearchTree<String>()
+
     let testCount = 30
     
     func testCanFindStringInTree() {
         var testStrings: [(key: String, data: String)] = []
+        let treeOfStrings = TernarySearchTree<String>()
+
         for _ in (1...testCount) {
```

Comment:
I see you point but...

The first test: `testCanFindStringInTree`, generates a tree containing `String` elements so therefore the `testStrings` and `treeOfStrings` are created there, like so:

        var testStrings: [(key: String, data: String)] = []
        let treeOfStrings = TernarySearchTree<String>()

The second test: `testCanFindNumberInTree`, generates a tree containing `Int` elements, so here the variables used looks like this:

        var testNums: [(key: String, data: Int)] = []
        let treeOfInts = TernarySearchTree<Int>()

(notice that one tree works on `String` elements and the other works on `Int` elements).

If all tests were using the same tree structure then I agree that it would be way better to generate/populate the `treeOfStrings` and the `testStrings` in `setup()`, but now the two tests are using different trees for their testing, and I think adding them as two different properties on the testcase would clutter the testclass.

If we add more tests at a later point then perhaps we should consider extracting `treeOfStrings` into a property and then do tests only with `String` elements.

Hope this makes sense to you (not trying to dodge the task here 😄)